deskreen: init at 2.0.4

leo248 opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,76 @@
+  pango
+}:
+
+stdenv.mkDerivation rec{
```
I think we should use the `finalAttrs` pattern here.

The `finalAttrs` pattern will let you remove the `rec` keyword ([see its implementation in Nix](https://github.com/NixOS/nix/blob/2d1d81114d72ace89ce08cd3bc93f4eb27a2975d/src/libexpr/nixexpr.cc#L121)).

Why is this pattern preferred to `rec` ?

Let's take this simple code example:

```nix
mkDerivation rec {
   foo = 1;
   bar = foo + 1;
}
```

and then `.overrideAttrs(old: { foo = 2; })`, you'll get `{ foo = 2; bar = 2; }` while with `finalAttrs` pattern, it would work correctly because it's a real fixed point.

Let me share a couple of useful links regarding the `finalAttrs` pattern:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Feel free to reach out if you need some assistance.

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,76 @@
+  pname = "deskreen";
+  version = "2.0.4";
+
+  deskreen = appimageTools.wrapType2{
```
```suggestion
  deskreen = appimageTools.wrapType2 {
```

Please be consistent with code style. Try to make sure to add a space before the function call and its parameter.

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,76 @@
+    name = pname;
+    src = fetchurl {
+      url = "https://github.com/pavlobu/deskreen/releases/download/v${version}/Deskreen-${version}.AppImage";
+      sha256 = "sha256-0jI/mbXaXanY6ay2zn+dPWGvsqWRcF8aYHRvfGVsObE=";
```
```suggestion
      hash = "sha256-0jI/mbXaXanY6ay2zn+dPWGvsqWRcF8aYHRvfGVsObE=";
```

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,76 @@
+      url = "https://github.com/pavlobu/deskreen/releases/download/v${version}/Deskreen-${version}.AppImage";
+      sha256 = "sha256-0jI/mbXaXanY6ay2zn+dPWGvsqWRcF8aYHRvfGVsObE=";
+    };
+    extraPkgs = pkgs: with pkgs;[
```
```suggestion
    extraPkgs = pkgs: with pkgs; [
```

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,76 @@
+    dontUnpack = true;
+    dontBuild = true;
+
+    installPhase = ''
```
If you are overriding `configurePhase`, `buildPhase`, `checkPhase`, `installPhase` or any other phase, you should not forget about explicitly running pre and post-phase hooks like [their](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L953) [original](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1008) [definitions](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1037) [do](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1078).

It is generally expected that an appropriate pre-phase hook (e.g. `preBuild`) hook will run at the beginning of a phase (e.g. `buildPhase`) and post-phase hook (e.g. `postBuild`) will run at the end. Hooks are normally ran as a part of a phase so if you override a phase as a whole, you will need to add `runHook hookName` calls (e.g. `runHook preBuild`) manually.

Having phases run pre/post-phase hooks is important because many setup hooks insert their own code into them – omitting a hook might therefore prevent some setup hooks required for proper functionality of a package from running. Additionally, hooks are often inserted by developers into the package expression and by users when overriding a package using `overrideAttrs`. Not running them can thus cause confusion why their code is not executed.

## Examples

### Before

```nix
  installPhase = ''
    your commands
  '';
```

### After

```nix
  installPhase = ''
    runHook preInstall

    your commands

    runHook postInstall
  '';
```

Source: https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/missing-phase-hooks.md

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,76 @@
+      cp ${deskreen}/bin/* $out/bin/
+    '';
+
+    meta = with lib; {
+    license = licenses.agpl3;
+    description = "Deskreen - Turn any device into a secondary screen for your computer";
+    homepage = "https://deskreen.com";
+    maintainers = with maintainers; [ leo248 ];
+    platforms = platforms.unix;
+    };
```
Please, fix indentation and sort attributes alphabetically.

drupol reviewed (changes requested):
Hi !

Thanks for your contribution.

I've left some comments, let me know if this is clear enough.

Also, make sure your PR contains only 2 commits:

1. One commit to add yourself as maintainer: `maintainers: add leo248`
2. The second commit for the rest of the changes: `deskreen: init at 2.0.4`

Make sure this PR title match the commit log message: `deskreen: init at 2.0.4`

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,83 @@
+, xorg
+}:
+
+stdenv.mkDerivation (finalAttrs: {
```
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
```

You can safely use `stdenvNoCC` since you don't need to compile anything in here.

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,83 @@
+  deskreen = appimageTools.wrapType2 {
+    name = pname;
+    src = fetchurl {
+      url =
+        "https://github.com/pavlobu/deskreen/releases/download/v${version}/Deskreen-${version}.AppImage";
```
```suggestion
      url = "https://github.com/pavlobu/deskreen/releases/download/v${version}/Deskreen-${version}.AppImage";
```

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,83 @@
+  version = "2.0.4";
+
+  deskreen = appimageTools.wrapType2 {
+    name = pname;
```
```suggestion
    name = "deskreen";
```

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,83 @@
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Deskreen - Turn any device into a secondary screen for your computer";
+    homepage = "https://deskreen.com";
+    license = licenses.agpl3;
+    maintainers = with maintainers; [ leo248 drupol ];
+    platforms = platforms.unix;
+  };
```
Missing `meta.mainProgram`

drupol reviewed (changes requested):
Hi!

Good, this is your first pull request, thanks for it!

I made further comments, let me know if you need some help, I'm here to help.

drupol commented:
@leo248 Why did you closed the PR ?

leo248 commented:
Hi @drupol,
I closed the PR due to build issues and my limited availability for the next 14 days, but your interest has prompted me to reconsider. If you're willing to help resolve these issues, I'd be happy to reopen it for collaboration

drupol commented:
Sure reopen it and I'll complete it !

leo248 committed:
deskreen: init at 2.0.4

Update pkgs/by-name/de/deskreen/package.nix

Co-authored-by: Anderson Torres <torres.anderson.85@protonmail.com>

Update pkgs/by-name/de/deskreen/package.nix

Co-authored-by: Anderson Torres <torres.anderson.85@protonmail.com>

Update pkgs/by-name/de/deskreen/package.nix

leo248 commented:
Thank you for all the assistance. I'm about to push my latest changes, as it seems everything is functioning correctly now. Your support has been invaluable. I'll be taking a break for two weeks, but when I return, I'm looking forward to contributing more effectively. Thank you again for the helpful introduction to this process.

drupol commented:
@GrahamcOfBorg build deskreen

drupol reviewed (approved):
All good, thanks for your contribution, keep them coming ! :)

AndersonTorres commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,89 @@
+  '';
+
+  meta = {
+    description = "Deskreen - Turn any device into a secondary screen for your computer";
```
```suggestion
    description = "Turn any device into a secondary screen for your computer";
```

No self-reference on description

AndersonTorres commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,89 @@
+      nspr
+      nss
+      pango
+      xorg.libX11
+      xorg.libxcb
+      xorg.libXcomposite
+      xorg.libXdamage
+      xorg.libXext
+      xorg.libXfixes
+      xorg.libXrandr
+      xorg.libXrender
+      xorg.libXtst
```
```suggestion
      libX11
      libxcb
      libXcomposite
      libXdamage
      libXext
      libXfixes
      libXrandr
      libXrender
      libXtst
```

AndersonTorres commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,89 @@
+  deskreenUnwrapped = appimageTools.wrapType2 {
+    name = "deskreen";
+    src = finalAttrs.src;
+    extraPkgs = pkgs: with pkgs; [
```
Why?

h7x4 replied:
It's not obvious what you're asking when you just point at a line and say *why*. Is it the `with` statement you're hinting at?

drupol replied:
Good suggestion. Thanks.
What it is advised to do in this case then? What do you advise?

AndersonTorres replied:
I know nothing about appimageTools, but looking at the Nixpkgs, I have seen many of  them that declare only lib, stdenv, fetchers and appimageTools in the input set; when they use extraPackages, they do not declare them on the input set.

So, my suggestion (with big grains of salt) would be remove them from input attrset.

drupol commented on pkgs/by-name/de/deskreen/package.nix:
```diff
@@ -0,0 +1,43 @@
+{ lib,
+  fetchurl,
+  appimageTools,
+  stdenvNoCC }:
```
This is what we usually do:

```suggestion
{ lib
, fetchurl
, appimageTools
, stdenvNoCC
}:
```

leo248 committed:
maintainers: add leo248

Update pkgs/by-name/de/deskreen/package.nix

Co-authored-by: Pol Dellaiera <pol.dellaiera@protonmail.com>

	modified:   maintainers/maintainer-list.nix

drupol reviewed (changes requested):
LGTM now, thanks for fixing everything.

Could you please squash your commits now?

There must be 2 commits only in this PR:

1. maintainers: add `leo248`
2. deskreen: init at 2.0.4

h7x4 commented:
Do you think you could sign those commits with your pgp key as well? I don't think commits needs to be signed in general, but [we can't verify the keys unless at least the first commit is signed](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md#individual-maintainer-list).

drupol commented:
Thanks for your first contribution!

Keep PRs coming :)
